packages/devkit/src/utils/invoke-nx-generator.ts

```diff
@@ -0,0 +1,140 @@
+import { join } from 'path';
+import { FileChange } from '@nrwl/tao/src/shared/tree';
+import { Generator, GeneratorCallback } from '@nrwl/tao/src/shared/workspace';
+
+class RunCallbackTask {
+  constructor(private callback: GeneratorCallback) {}
+  toConfiguration() {
+    return {
+      name: 'RunCallback',
+      options: {
+        callback: this.callback,
+      },
+    };
+  }
+}
+
+function createRunCallbackTask() {
+  return {
+    name: 'RunCallback',
+    create: () => {
+      return Promise.resolve(
+        async ({ callback }: { callback: GeneratorCallback }) => {
+          await callback();
+        }
+      );
+    },
+  };
+}
+
+/**
+ * Convert an Nx Generator into an Angular Devkit Schematic
+ */
+export function convertNxGenerator<T = any>(generator: Generator<T>) {
+  return (options: T) => invokeNxGenerator(generator, options);
+}
+
+/**
+ * Create a Rule to invoke an Nx Generator
+ */
+function invokeNxGenerator<T = any>(generator: Generator<T>, options: T) {
+  return (tree, context) => {
+    if (context.engine.workflow) {
+      const engineHost = (context.engine.workflow as any).engineHost;
+      engineHost.registerTaskExecutor(createRunCallbackTask());
+    }
+
+    const adapterTree = new AdapterTree(tree);
+    const result = generator(adapterTree, options);
+
+    if (!result) {
+      return adapterTree['tree'];
+    }
+    if (typeof result === 'function') {
+      if (context.engine.workflow) {
+        context.addTask(new RunCallbackTask(result));
+      }
+    }
+  };
+}
+
+const actionToFileChangeMap = {
+  c: 'CREATE',
+  o: 'UPDATE',
+  d: 'DELETE',
+};
+
+class AdapterTree {
```

Comment:
I'd rename it to show the direction of the wrapping.  The reason is that we have so many things wrapping other things to make this integration seamless that I get lost. Something like NxDevkitTreeWrappingAngularDevkitTree.